### PR TITLE
[chore] Update compatibility to a recent Triton

### PR DESCRIPTION
**Pull request by blefaudeux**
## What does this PR do?
- Moves all the triton layers to a more recent release of OpenAI Triton. Not all of them make sense by now, given nvfuser or pytorch perf improvements, dropping some of them would be fine by me but at least this PR will make an xformer install alongside some recent triton code easier
- Updates all the plots to reflect a newer pytorch and a newer triton
- Added the StarReLU activation (from Metaformer baselines), because why not, check extensibility

## Before submitting
- [x] Did you have fun?
  - Make sure you had fun coding 🙃
- [x] Did you read the [contributor guideline](https://github.com/facebookresearch/xformers/blob/master/CONTRIBUTING.md)?
- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
  - [x] N/A
- [x]  Did you make sure to update the docs?
  - [ ] N/A
- [x] Did you write any new necessary tests?
  - [ ] N/A
- [x] Did you update the [changelog](https://github.com/facebookresearch/xformers/blob/master/CHANGELOG.md)? (if needed)
  - [ ] N/A

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

**Comment by danthe3rd** [comment]
Oh that's great! Thanks for the contribution! I gave it a try in https://github.com/facebookresearch/xformers/pull/521 by simply changing the requirements.txt file, but it looks like there is more to it

**Comment by blefaudeux** [comment]
> Oh that's great! Thanks for the contribution! I gave it a try in #521 by simply changing the requirements.txt file, but it looks like there is more to it

yes, some of the interface changed (you cannot compose kernels anymore for instance). It should work with more recent versions I believe, this is post-breaking changes

File: xformers/triton/k_dropout.py [hunk 1]
```diff
@@ -18,66 +31,8 @@
...
-    # optional: fused activation (while the data is in shared memory)
-    if ACTIVATION:
-        x = ACTIVATION(x)
-
-    # randomly prune (and scale) the resulting buffer, possibly a no-op
-    output = _random_prune_and_scale(x, rand_mask, p, p_scale)
-
-    tl.store(y_ptrs, output, mask=block_mask)  # output
+MAX_INT32 = 2147483647
+MAX_UINT32 = 4294967295
```
Comment (danthe3rd) [inline xformers/triton/k_dropout.py:35]: nit: maybe use this for readability?
```suggestion
MAX_INT32 = (1 << 31) - 1
MAX_UINT32 = (1 << 32) - 1
```

**Review by danthe3rd — COMMENTED** [review]
Thanks a lot! Happy to review/merge once you think it's ready :)

**Comment by danthe3rd** [comment]
> ouch, the latest update doesn't work on a T4 used for CI, too old.. I'll fix that later when I get the time :(

Do you think we can merge it without the latest changes? As this is currently blocking users from installing xformers with triton at all

**Comment by blefaudeux** [comment]
> > ouch, the latest update doesn't work on a T4 used for CI, too old.. I'll fix that later when I get the time :(
> 
> Do you think we can merge it without the latest changes? As this is currently blocking users from installing xformers with triton at all

it's just a dimension preset on the fused linear kernel, I'll update that

**Review by danthe3rd — APPROVED** [review]
Thanks a lot for maintaining the triton part of the code and updating it!
Really appreciated :)

**Pull request merged**